server/src/main/java/org/opensearch/action/admin/indices/tiering/RestWarmTieringAction.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.tiering;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.core.common.Strings.splitStringByCommaToArray;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * Rest Tiering API action to move indices to warm tier
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class RestWarmTieringAction extends BaseRestHandler {
+
+    private static final String TARGET_TIER = "warm";
+
+    @Override
+    public List<RestHandler.Route> routes() {
+        return singletonList(new RestHandler.Route(POST, "/{index}/_tier/" + TARGET_TIER));
```

Comment:
nit: is it better to rename it `index_pattern`?

Reply:
i will leave it as is following the convention used in other index apis that use index pattern as well